### ISSUE TITLE
React PropTypes and createClass warnings

### ISSUE THREAD

**Issue by AlexJF** [erikras/redux-form#3385]
### Are you submitting a **bug report** or a **feature request**?
Bug report

### What is the current behavior?
When running jest unit tests touching redux-form related code, those tests log React warnings to the console:

```
console.warn node_modules\react\lib\lowPriorityWarning.js:40
      Warning: Accessing PropTypes via the main React package is deprecated, and will be removed in  React v16.0. Use the latest available v15.* prop-types package from npm instead. For info on usage, compatibility, migration and more, see https://fb.me/prop-types-docs
    console.warn node_modules\react\lib\lowPriorityWarning.js:40
      Warning: Accessing createClass via the main React package is deprecated, and will be removed in React v16.0. Use a plain JavaScript class instead. If you're not yet ready to migrate, create-react-class v15.* is available on npm as a temporary, drop-in replacement. For more info see https://fb.me/react-create-class
```

### What is the expected behavior?
No warnings :)

### What's your environment?
Redux-form 7.0.4
Babel 6.26.0
Jest 21.0.1
Flow 0.49.1
React 15.6.1

### Other information
Related issue: https://github.com/facebook/flow/issues/4673

Culprits: https://github.com/erikras/redux-form/search?l=JavaScript&q=%22+import+%2A+as+React%22+&type=&utf8=%E2%9C%93

**Comment by slavab89** [comment]
I actually see this as well, but on my SSR build.
This happens since `7.0.4`. I dont see this message on `7.0.3`

**Comment by gustavohenke** [comment]
Hi folks, thanks for the report!
This really seems to be code added as per v7.0.4 release. Please see #3387 for a potential fix 😉

**Comment by erikras** [comment]
I saw this during my tests, but could not figure out where it was coming from, as I was not specificaly importing `PropTypes` from anywhere. Good work by @gustavohenke on the fix.

The `import * from React` was taken directly from the Flow docs.

**Comment by erikras** [comment]
Published fix in [`v7.1.0`](https://github.com/erikras/redux-form/releases/tag/v7.1.0).

**Comment by nerfologist** [comment]
Hello @erikras, unfortunately I'm still experiencing these warnings in Jest after upgrading to `v7.1.0`. Grepping through the `v7.1.0` version I found the following imports which may be the culprits:
```
✎  $ ag "from 'react'" | grep '*'
src/FieldArrayProps.types.js.flow:2:import * as React from 'react'
src/FieldProps.types.js.flow:4:import * as React from 'react'
src/FieldsProps.types.js.flow:2:import * as React from 'react'
src/immutable.js.flow:2:import * as React from 'react'
src/index.js.flow:2:import * as React from 'react'
```
I was able to confirm that this import causes the yellow warning in Jest:
```
  console.warn node_modules/react/lib/lowPriorityWarning.js:40
    Warning: Accessing PropTypes via the main React package is deprecated, and will be removed in  React v16.0. Use the latest available v15.* prop-types package from npm instead. For info on usage, compatibility, migration and more, see https://fb.me/prop-types-docs

  console.warn node_modules/react/lib/lowPriorityWarning.js:40
    Warning: Accessing createClass via the main React package is deprecated, and will be removed in React v16.0. Use a plain JavaScript class instead. If you're not yet ready to migrate, create-react-class v15.* is available on npm as a temporary, drop-in replacement. For more info see https://fb.me/react-create-class
```
HTH, thanks for your amazing work!
